src/devices/video/EmuNV2A_PGRAPH.cpp

```diff
@@ -4804,66 +4804,9 @@ int pgraph_handle_method(
-								// ** is the sequence of call to NV097_SET_TRANSFORM_EXECUTION_MODE and NV097_SET_TRANSFORM_PROGRAM_START
-								// ** program uses D3DDevice_SelectVertexShader() which calls NV097_SET_TRANSFORM_EXECUTION_MODE first then calls NV097_SET_TRANSFORM_PROGRAM_START
-								// ** passthrough in SelectVertexShader() calls NV097_SET_TRANSFORM_PROGRAM_START first, then calls NV097_SET_TRANSFORM_EXECUTION_MODE
-
-
                                 g_NV2A_VertexShaderMode = VertexShaderMode::ShaderProgram;
-                                //g_UseFixedFunctionVertexShader = false;
-
-                                // for shader program, here we set it to default register 0, later when we reach NV097_SET_TRANSFORM_PROGRAM_START, we'll use the register addr passed in.
-                                //g_Xbox_VertexShader_FunctionSlots_StartAddress = 0;
-
-                                // set vertex shader dirty flag
                                 g_NV2AVertexShader_dirty = true;
 
-                            }
-                            else {
-                                // if we hit here with g_Xbox_VertexShaderMode==FixedFunction, then we're in Passthrough
-                                //if (g_VertexShader_dirty == false) {
-
-                                    g_NV2A_VertexShaderMode = VertexShaderMode::Passthrough;
-                                    //g_UseFixedFunctionVertexShader = false;
-
-                                    // for shader program, here we set it to default register 0, later when we reach NV097_SET_TRANSFORM_PROGRAM_START, we'll use the register addr passed in.
-                                    g_NV2A_VertexShader_FunctionSlots_StartAddress = 0;
-
-                                    // set vertex shader dirty flag
-                                    g_NV2AVertexShader_dirty = true;
-
-                                    // funtion key F7 flips this variable
-                                    //g_bUsePassthroughHLSL = true;
-                                    //float tempConstant[4];
-                                    // read constant register 0, CommonSetPassThroughProgram() sets register 0 constant with SuperSampleScaleX/Y
-                                    //CxbxrImpl_GetVertexShaderConstant(0 - X_D3DSCM_CORRECTION, tempConstant, 1);
-                                    //extern void CxbxrSetSuperSampleScaleXY(float x, float y);
-                                    //CxbxrSetSuperSampleScaleXY(tempConstant[0], tempConstant[1]);
-                                //}
-                            }
-						}
-						/*
-						fix function setup finished using 
-						push (NV097_SET_TRANSFORM_EXECUTION_MODE,
-								 NV097_SET_TRANSFORM_EXECUTION_MODE_MODE_FIXED|NV097_SET_TRANSFORM_EXECUTION_MODE_RANGE_MODE_PRIV, method count 1)
-						to start fix function vertex shader program
-						*/
 					}else if (slot== NV097_SET_TRANSFORM_EXECUTION_MODE_MODE_FIXED){//fix function mode
```

Comment:
please change line 4810 from 
}else if (slot== NV097_SET_TRANSFORM_EXECUTION_MODE_MODE_FIXED){//fix function mode
to 
}else{
  we don't need the test for  if (slot== NV097_SET_TRANSFORM_EXECUTION_MODE_MODE_FIXED) here any more.

Reply:
done....